src/WorkerChannel.ts

```diff
@@ -119,7 +119,15 @@ export class WorkerChannel implements IWorkerChannel {
     public workerInitRequest(requestId: string, msg: rpc.WorkerInitRequest) {
         // Validate version
         const version = process.version;
-        if (!(version.startsWith('v14.') || version.startsWith('v16.'))) {
+        if (version.startsWith('v17.') && process.env.AZURE_FUNCTIONS_ENVIRONMENT == 'Development') {
+            const msg =
+                'Node.js version used (' +
+                version +
+                ') is not officially supported. You may use it during local development, but your function will be deployed to a different Node.js version on Azure.' +
+                ' The version of the Azure Functions runtime you are using (v4) officially supports Node.js v14.x or Node.js v16.x.' +
+                ' Refer to our documentation to see the Node.js versions supported by each version of Azure Functions: https://aka.ms/functions-node-versions';
+            systemWarn(msg);
+        } else if (!(version.startsWith('v14.') || version.startsWith('v16.'))) {

```

Comment:
Huh...any thoughts on why we have duplicated logic to check the version in this file and the other?

My hunch is that it was only because of the "_v1WorkerBehavior" flag on the v2.x branch: https://github.com/Azure/azure-functions-nodejs-worker/blob/v2.x/src/WorkerChannel.ts#L131 and that we can get rid of the logic in this file since v3.x branch doesn't have any backwards compat flag like that.

Reply:
I'm not sure why there is this double logic, but it is needed on this branch too. I only modified this file because during my tests, function apps with node version ~17 would error out here.

Reply:
Yeah I agree you'll probably need to adjust the logic in both places, but let's still try to avoid duplicating as much as we can. For example, we want to make sure the warning is only printed once

Reply:
Yeah I agree with you. I just adjusted the code so that here it's a gRPC log (makes it to the host) and the other one is a node process log

Reply:
I'm concerned that the only reason these logs are currently separate is because we're running in a development environment with our local worker build. I worry once we integrate a real version of the node worker into the core tools, it'll show the warning twice in one place. Are you confident that won't happen? I just don't know enough about how the logs work to be sure.

Otherwise, we could do something like this in `nodejsWorker.ts`:
```
} else if (process.env.AZURE_FUNCTIONS_ENVIRONMENT == 'Development' && devOnlyVersions.indexOf(major) >= 0) {
    // Let users continue here. Warning will be shown in the workerInitRequest
}
```

Reply:
I'm fairly confident that it won't happen. I tried starting my function app from core tools without specifying a local worker directory, and I didn't get any logs from the worker's console, which I would expect to see (e.g. `systemLog(`Worker ${workerId} connecting on ${connection}`);` or `systemLog(`Worker ${workerId} exited with code ${code}`);` in `Worker.ts`. Also, when I trigger an exception by using `v17.x` of node, I don't see the `logPreix = "LanguageWorkerConsoleLog"` at the beginner of the error message, indicating the error in the core tools is coming from `throw` statement, not from the `console.error`.

Reply:
Thanks for the investigation! It felt wrong that we weren't seeing any of those logs, so I looked into it further. Turns out you can view them by adding this to your local.settings.json: `"logging:logLevel:Worker": "debug"` (more details [here](https://github.com/anthonychu/functions-log-suppression)). I think it's a bug that all worker console logs are currently categorized as "Debug" so I submitted this PR to fix: https://github.com/Azure/azure-functions-host/pull/8102

Assuming that gets merged, the console warnings _would_ be displayed by default

Reply:
Thanks for looking into this! I confirmed this by adding that setting to my local.settings.json, and indeed the warning message was displayed twice (although the one from `console.warn` was not yellow or clear at all that it was a warning -- hopefully this gets fixed by your PR). I went ahead and removed this duplicate warning. Good find!